exo/main.py

```diff
@@ -225,6 +230,23 @@ def handle_exit():
 
   await node.start(wait_for_peers=args.wait_for_peers)
 
+  # Preload models if specified
+  if args.preload_models:
+    models_to_preload = [model.strip() for model in args.preload_models.split(",")]
+    print(f"Preloading models: {models_to_preload}")
+    for model_name in models_to_preload:
+      try:
+        shard = build_base_shard(model_name, inference_engine.__class__.__name__)
+        if not shard:
+          print(f"Warning: Unsupported model '{model_name}' for inference engine {inference_engine.__class__.__name__}")
+          continue
+        await node.preload_models([shard])
```

Comment:
Shouldn't we build all the shards first then pass it into here. I think this could be much shorter with some list comprehensions and filtering.

Reply:
can you check the implementation now ? does it suit your requirements ?